qccalculator/cli.py

```diff
@@ -11,151 +10,252 @@
+  rqs.append(rq)
+
+  global_variables(output, zip)
+  finale()
+
+
+@click.command("maxq", short_help="Compute the MaxQuant quality metrics")
+@click.option('--zipurl', type=click.Path(exists=True), required=True,
+              help="The URL to a max quant output zip file (must contain evidence.txt and parameters.txt).")
+@click.option('--rawname', type=str, default="",
+              help="The raw file name of interest (as in evidence.txt) without path or extension.")
+@common_options
+def maxq(zipurl, rawname, mzml, output, zip):
+  """
+  Calculate all possible metrics id and spectra for MaxQuant output. These data sources will be included in set metrics.
+
+  Parameters
+  ----------
+  zipurl: zip file with all the MQ outputs.
+  rawname: The files that will be use to compute the QC metrics.
+  mzml: the mzML files to compute the QC metrics
+
+  Returns
+  -------
+
+  """
+  exp = oms.MSExperiment()
+  oms.MzMLFile().load(click.format_filename(mzml), exp)
+  rq = basicqc.getBasicQuality(exp)
+
+  ms2num = 0
+  for x in rq.qualityMetrics:
+    if x.name == "Number of MS2 spectra":
+      ms2num = x.value
+
+  if ms2num < 1:
+    logging.warn("We seem to have found no MS2 spectra which is unlikely to be true since you have also given some identifications. \
                 We continue with symbolic value of 1 for the number of MS2 spectra, \
                 however this means some metrics will invariably be incorrect!\
                 Please make sure, we have the right inputs.")
-        ms2num = 1
+    ms2num = 1
 
-    try:
-        mq,params = idqcmq.loadMQZippedResults(zipurl)
-        if not rawname:
-            logging.warning("Infering rawname from mzML")
-            rawname = basename(exp.getExperimentalSettings().getSourceFiles()[0].getNameOfFile().decode()) # TODO split extensions
+  try:
+    mq, params = idqcmq.loadMQZippedResults(zipurl)
+    if not rawname:
+      logging.warning("Inferring rawname from mzML")
+      rawname = basename(
+        exp.getExperimentalSettings().getSourceFiles()[0].getNameOfFile().decode())  # TODO split extensions
 
-        rq.qualityMetrics.extend(idqcmq.getMQMetrics(rawname, params,mq, ms2num))
-        rqs.append(rq)
-    except:
-        logging.warn("Retrieving any results from the URL failed.")
+    rq.qualityMetrics.extend(idqcmq.getMQMetrics(rawname, params, mq, ms2num))
+    rqs.append(rq)
+  except:
+    logging.warn("Retrieving any results from the URL failed.")
 
-    finale()
+  global_variables(output, zip)
+  finale()
 
 
-@start.command()
-@click.argument('filename', type=click.Path(exists=True))
-def basic(filename):
-    """Calculate the basic metrics available from virtually every mzML file."""
-    exp = oms.MSExperiment()
-    oms.MzMLFile().load(click.format_filename(filename), exp)
-    rq = basicqc.getBasicQuality(exp)
-    rqs.append(rq)
+@click.group(context_settings=CONTEXT_SETTINGS)
+def cli():
```

Comment:
qccalculator basic [OPTIONS] is: 
```
...
--mzml TEXT 
```
I think it should be `--mzml PATH [required]` 
so we dont need to check the argument before trying to open the file.

Reply:
problem solved